src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,37 @@
+import Popup from "./Popup.js";
+
+export class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleConfirm) {
+    super(popupSelector);
+    this._handleConfirm = handleConfirm.bind(this);
+    this._data = null;
+  }
+
+  open(data) {
+    super.open(data);
+    this._data = data;
+    console.log("Opening confirm popup with data:", data);
+    const confirmButton = this._popupElement.querySelector(
+      ".modal__confirm-button"
+    );
+    if (confirmButton) {
+      confirmButton.removeEventListener("click", this._handleConfirmBound);
+      this._handleConfirmBound = () => {
+        if (this._data && this._data.cardId) {
+          this._handleConfirm(this._data);
+        } else {
+          console.error("No cardId in popup data:", this._data);
+        }
+      };
+      confirmButton.addEventListener("click", this._handleConfirmBound);
+    }
+  }
+  close() {
+    super.close();
+    this._data = null;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.